### ISSUE TITLE
How to use sd_setImage completed block with GIFs?

### ISSUE THREAD

**Issue by JEKos** [rs/SDWebImage#2073]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 11
 SDWebImage Version      | 4.1.2
 Integration Method      | cocoapods
 Xcode Version           | Xcode 9
 Repro rate              |  all the time (100%)

### Issue Description and Steps

Hi,
I'm trying to implement sd_setImage with completedBlock so the GIFs will load with animation only once when load the collection view and scroll back to the same GIFs. The follow method works with photos but not with GIFs because the completedBlock returns UIImage and not FLAnimatedImage that I need in this situation:

```
MyFLAnimatedImageView.sd_setImage(with: URL(string: gifsarray[indexPath.row]) { (image, error, imageCacheType, imageUrl) in
                if imageUrl == (URL(string: self.gifsarray[indexPath.row])) {
                    if imageCacheType == SDImageCacheType.memory {
                        MyFLAnimatedImageView.image = image
                    } else {
                        UIView.transition(with: MyFLAnimatedImageView, duration: 0.3, options: .transitionCrossDissolve, animations: {() -> Void in
                            MyFLAnimatedImageView.image = image
                            activityIndicator.removeFromSuperview()
                        }, completion: { _ in })
                    }
                    if image != nil {
                        activityIndicator.removeFromSuperview()
                    }
                    else
                    {
                        print("image not found")
                    }
                }
            }
```

**Comment by bpoplauschi** [comment]
@JEKos please fill in the info required (as in the issue template). I don't want to take it on you, but why are you ignoring that stuff that is already pre-filled and you just have to spend a few seconds to fill it in.

**Comment by JEKos** [comment]
Ok, done

**Comment by bpoplauschi** [comment]
@JEKos Unfortunately, `FLAnimatedImage` is not a `UIImage`, so it doesn't work with all the APIs we have in place. We had a conversation with `FLAnimatedImage` team about this, but they only created a proof of concept and never integrated it (to make `FLAnimatedImage` a `UIImage` subclass).
I don't see how we could return the `FLAnimatedImage` there.

**Comment by JEKos** [comment]
I tried with SDImageFormat to convert it from UIImage to FLAnimatedImage but it doesn't work too

**Comment by bpoplauschi** [comment]
The only way you could convert from `UIImage` to `FLAnimatedImage` would be to transform the `UIImage` to `NSData` and then call `[FLAnimatedImage animatedImageWithData:]`

**Comment by JEKos** [comment]
Problem solved!
Just called again `MyFLAnimatedImageView.sd_setImage(with: imageUrl)` in both times, instead of `MyFLAnimatedImageView.image = image`
